sdk/tables/data-tables/test/integration/utils/recordedClient.ts

```diff
@@ -47,53 +47,104 @@ export const recordedEnvironmentSetup: RecorderEnvironmentSetup = {
-    );
-  }
+    case "SASToken":
+      if (!env.SAS_TOKEN || !env.TABLES_URL) {
+        throw new Error(
+          "SAS Token and AccountURL must be defined, make sure that SAS_TOKEN and  TABLES_URL are defined in the environment"
+        );
+      }
 
-  throw new Error(
-    "Couldn't find Connection String or Account Key, Name and SAS in the environment variables"
-  );
-}
+      return new TableClient(`${env.TABLES_URL}${env.SAS_TOKEN}`, tableName);
 
-export function createTableServiceClient(): TableServiceClient {
-  if (env.STORAGE_CONNECTION_STRING) {
-    return TableServiceClient.fromConnectionString(env.STORAGE_CONNECTION_STRING);
-  }
+    case "AccountKey":
+      if ((!env.ACCOUNT_NAME && !env.ACCOUNT_KEY) || !env.TABLES_URL) {
+        throw new Error(
+          "AccountName, AccountURL and AccountKey must be defined, make sure that ACCOUNT_NAME, ACCOUNT_KEY and TABLES_URL are defined in the environment"
+        );
+      }
 
-  if (!env.ACCOUNT_URL) {
-    throw new Error("Either STORAGE_CONNECTION_STRING or ACCOUNT_URL must be defined");
-  }
+      return new TableClient(
+        env.TABLES_URL,
+        tableName,
+        new TablesSharedKeyCredential(env.ACCOUNT_NAME, env.ACCOUNT_KEY)
+      );
 
-  if (env.ACCOUNT_SAS) {
-    return new TableServiceClient(`${env.ACCOUNT_URL}${env.SAS_TOKEN}`);
-  }
+    case "AccountConnectionString":
+      if (!env.ACCOUNT_CONNECTION_STRING) {
+        throw new Error(
+          "AccountConnectionString is not defined, make sure that ACCOUNT_CONNECTION_STRING is defined in the environment"
+        );
+      }
 
-  if (env.ACCOUNT_NAME && env.ACCOUNT_KEY) {
-    return new TableServiceClient(
-      env.ACCOUNT_URL,
-      new TablesSharedKeyCredential(env.ACCOUNT_NAME, env.ACCOUNT_KEY)
-    );
+      return TableClient.fromConnectionString(env.ACCOUNT_CONNECTION_STRING, tableName);
+
+    default:
+      throw new Error(`Unknown authentication mode ${mode}`);
   }
+}
 
-  throw new Error(
-    "Couldn't find Connection String or Account Key, Name and SAS in the environment variables"
-  );
+export function createTableServiceClient(mode: CreateClientMode = "SASConnectionString"): TableServiceClient {
+  switch (mode) {
+    case "SASConnectionString":
+      if (!env.SAS_CONNECTION_STRING) {
+        throw new Error(
+          "SASConnectionString is not defined, make sure that SAS_CONNECTION_STRING is defined in the environment"
+        );
+      }
+
+      return TableServiceClient.fromConnectionString(env.SAS_CONNECTION_STRING);
+
+    case "SASToken":
+      if (!env.SAS_TOKEN || !env.TABLES_URL) {
+        throw new Error(
+          "SAS Token and AccountURL must be defined, make sure that SAS_TOKEN and  TABLES_URL are defined in the environment"
+        );
+      }
+
+      return new TableServiceClient(`${env.TABLES_URL}${env.SAS_TOKEN}`);
+
+    case "AccountKey":
+      if ((!env.ACCOUNT_NAME && !env.ACCOUNT_KEY) || !env.TABLES_URL) {
```

Comment:
Should this be `!env.ACCOUNT_NAME || !env.ACCOUNT_KEY || !env.TABLES_URL` instead?